src/derive/field.rs

```diff
@@ -68,83 +68,67 @@ macro_rules! field_common {
                 // If `val` represents a 256 bit value then `r` should be R^2,
                 // if `val` represents the 256 MSB of a 512 bit value, then `r` should be R^3.
 
-                #[cfg(feature = "asm")]
```

Comment:
I can't see that it would effect. Single implementation that is always remain constant (even with asm feature) would be enough. @sragss Can you confirm?

Reply:
#134 just ported the `asm` logic from prior. IIRC without the feature gated `asm` a few of the tests were failing due to some assumptions about the input. I believe this PR also removes those tests.

Reply:
I've just imported hash to curve test vectors for bn256 g1 and g2 from gnark-crypto. Those test runs in both asm enabled and disabled. So `test_consistent_hash_to_curve` can be safely removed now @davidnevadoc